pkg/providers/amifamily/resolver.go

```diff
@@ -34,6 +34,10 @@ import (
 	"sigs.k8s.io/karpenter/pkg/scheduling"
 )
 
+type AMIResolver interface {
```

Comment:
Following the naming convention for the other provider interfaces, this should be `Resolver` and we should rename the concrete type `DefaultResolver`.